File: Angular/src/app/shared/models/Usuario.ts
export class Usuario {
  private _token: string;
  private _tokenExp: Date;
  constructor(
    public email: string,
    public nome: string,
    public urlImagem?: string,
    public id?: number,
    public role?: 'adm' | 'user'
    ) { }

  set token(jwt: string) {
    this._token = jwt;
    const tokenInfo = JSON.parse(atob(jwt.split('.')[1]));
    this._tokenExp = new Date(+tokenInfo.exp * 1000);
  }

  get token() {
    if (new Date() > this._tokenExp) {
      return null;
    }
    return this._token;
  }

  get isAdm() {
    if (this.role === 'adm')
      return true;
    else
      return false;
  }

}

File: Models/Usuario.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using Maia.Models.Abstracts;
using Newtonsoft.Json;
using Maia.Models.DTO;
using Maia.Services;

namespace Maia.Models
{
    
    public class Usuario : Entidade<int>
    {
        [Required]
        [EmailAddress]
        public string Email { get; set; }

        [Required]
        [MinLength(3)]
        public string Nome { get; set; }

        [Required]
        [MinLength(6)]
        [DataType(DataType.Password)]
        [JsonIgnore]
        public string Senha { get; set; }

        [Range(typeof(string), "adm", "user")]
        public string Role { get; set; } = "user";
        
        public string UrlImagem { get; set; }
        
        public Usuario() { }

        public Usuario(UsuarioDTO dto, HasherService service)
        {
            this.Email = dto.Email;
            this.Nome = dto.Nome;
            this.Senha = service.HashPassword(dto.Senha);
        }
    }
}